mono/metadata/appdomain.c

```diff
@@ -987,12 +989,21 @@ mono_domain_owns_vtable_slot (MonoDomain *domain, gpointer vtable_slot)
 gboolean
 mono_domain_set (MonoDomain *domain, gboolean force)
 {
+	gboolean result;
+	MONO_ENTER_GC_UNSAFE;
```

Comment:
I think the external-only `mono_domain_set` should still do the `if (!force && ...)` stuff before entering GC Unsafe mode.  And then delegate directly to `mono_domain_set_internal_with_options`.

ie: `mono_domain_set` and `mono_domain_set_fast` are identical except the latter does a GC transition in the case where we've actually got to do some work.